spec/index.bs

```diff
@@ -1894,6 +1900,111 @@ dictionary GPUVertexAttributeDescriptor {
+
+                1. For each resource |resource| in |stage|.{{GPUProgrammableStageDescriptor/module}}
+                    for the |stage|.{{GPUProgrammableStageDescriptor/entryPoint}} entry-point:
+
+                    1. If |resource|'s "group" decoration is not equal to |index|:
+
+                        1. Continue to the next loop iteration over resources.
+
+                    1. Let |binding| be a new {{GPUBindGroupLayoutBinding}}.
+                    1. Set |binding|.{{GPUBindGroupLayoutBinding/binding}} to |resource|'s "binding"
+                        decoration.
+                    1. Set |binding|.{{GPUBindGroupLayoutBinding/visibility}} to
+                        {{GPUShaderStage/VERTEX}} | {{GPUShaderStage/FRAGMENT}} |
+                        {{GPUShaderStage/COMPUTE}}.
+
+                        Issue: Have a set of all valid {{GPUShaderStage}} on the device.
+
+                    1. If |resource| is for a sampler binding:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/sampler}}.
+
+                    1. If |resource| is for a buffer binding:
+
+                        1. If |resource| is for a uniform buffer:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/uniform-buffer}}.
+
+                        1. If |resource| is for a storage buffer:
+
+                            1. If |resource| is read-only:
+
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                    {{GPUBindingType/readonly-storage-buffer}}.
+
+                            1. Else:
+
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                    {{GPUBindingType/storage-buffer}}.
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/visibility}} to
+                                    {{GPUShaderStage/FRAGMENT}} | {{GPUShaderStage/COMPUTE}}.
+
+                    1. If |resource| is for a texture binding:
+
+                        1. Set |binding|.{{GPUBindGroupLayoutBinding/textureComponentType}} to
+                            |resource|'s component type.
+                        1. Set |binding|.{{GPUBindGroupLayoutBinding/textureDimension}} to
+                            |resource|'s dimension.
+                        1. If |resource| is multisampled:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/multisampled}} to true.
+
+                        1. If |resource| is for a sampled texture:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/sampled-texture}}.
+
+                        1. If |resource| is for a storage texture:
```

Comment:
why is there no `If |resource| is read-only:` clause?

Reply:
https://github.com/gpuweb/gpuweb/pull/532 isn't merged yet.